### DISCUSSION TITLE
how to get value after the demo has been parsed

### DISCUSSION THREAD

**Discussion by MonsieurBibo** [saul/demofile#317] [Questions]
Hey, I would like to create functions to get common values (player's kills, deaths...). However, I only get undefined values because my listeners are created after parsing or I'm trying to fetch data before parsing. I can share code if you want.

Thanks in advance.

**Comment by MonsieurBibo** [comment]
Well, after thinking and searching for a bit, I have found I could just do everything asynchronous. I'm putting everything I need to get into a promise that resolves when the demo ends. However, this is highly unpractical because every function will need one parsing process or I must create a big function that returns absolutely everything. I might just write interesting information into a JSON file and search in it.

**Reply by fady223** [reply]
How to check if the demo ends i couldn't find out
